pkg/controller/grafanadashboard/grafana_client.go

```diff
@@ -72,8 +74,32 @@ func NewGrafanaClient(url, user, password string, timeoutSeconds time.Duration)
 	}
 }
 
+func (r *GrafanaClientImpl) GetOrCreateNamespaceFolder(namespace string) (folderID int64, err error) {
+	gapiClient, err := grafanaapi.New("", r.url) // get the Auth from somewhere?
```

Comment:
@pb82 any idea where I could get this auth from? this seems to be required to create folders on grafana

Reply:
best to use the `CreateOrUpdateDashboard` function as a reference: https://github.com/integr8ly/grafana-operator/blob/master/pkg/controller/grafanadashboard/grafana_client.go#L76

Reply:
username and password are passed when creating a new instance of the api client: https://github.com/integr8ly/grafana-operator/blob/master/pkg/controller/grafanadashboard/dashboard_controller.go#L329